File: 250407/자동차 단일 거래 이익 최대화하기/max-profit-of-single-car.cpp
#include <iostream>
#include <climits>
using namespace std;

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int n = 0;
    int arr[1000] = {0};
    int minPrice = INT_MAX;
    int maxProfit = 0;
    
    cin >> n;
    for(int i = 0; i < n; i++) {
        cin >> arr[i];
        if(arr[i] < minPrice) {
            minPrice = arr[i];
        }
        int profit = arr[i] - minPrice;
        if(profit > maxProfit) {
            maxProfit = profit;
        }
    }
    cout << maxProfit;
    return 0;
}

File: 250407/자동차 단일 거래 이익 최대화하기/max-profit-of-single-car.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int arr[] = new int[1000];
        int minPrice = Integer.MAX_VALUE;
        int maxProfit = 0;

        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
            if (arr[i] < minPrice) {
                minPrice = arr[i];
            }
            int profit = arr[i] - minPrice;
            if (profit > maxProfit) {
                maxProfit = profit;
            }
        }
        System.out.println(maxProfit);
        sc.close();
    }
}